packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -268,48 +268,49 @@ module.exports = (
       )
       if (cachedToc) {
         return cachedToc
-      } else {
-        const ast = await getAST(markdownNode)
-        const tocAst = mdastToToc(ast, appliedTocOptions)
-
-        let toc
-        if (tocAst.map) {
-          const addSlugToUrl = function(node) {
-            if (node.url) {
-              if (
-                _.get(markdownNode, appliedTocOptions.pathToSlugField) ===
-                undefined
-              ) {
-                console.warn(
-                  `Skipping TableOfContents. Field '${appliedTocOptions.pathToSlugField}' missing from markdown node`
-                )
-                return null
-              }
-              node.url = [
-                basePath,
-                _.get(markdownNode, appliedTocOptions.pathToSlugField),
-                node.url,
-              ]
-                .join(`/`)
-                .replace(/\/\//g, `/`)
-            }
-            if (node.children) {
-              node.children = node.children
-                .map(node => addSlugToUrl(node))
-                .filter(Boolean)
-            }
+      }
 
-            return node
+      const ast = await getAST(markdownNode)
+      const tocAst = mdastToToc(ast, appliedTocOptions)
+
+      let toc = ``
+      if (tocAst.map) {
+        const addSlugToUrl = function(node) {
+          if (node.url) {
+            if (
+              _.get(markdownNode, appliedTocOptions.pathToSlugField) ===
+              undefined
+            ) {
+              console.warn(
+                `Skipping TableOfContents. Field '${appliedTocOptions.pathToSlugField}' missing from markdown node`
+              )
+              return null
+            }
+            node.url = [
+              basePath,
+              _.get(markdownNode, appliedTocOptions.pathToSlugField),
+              node.url,
+            ]
+              .join(`/`)
+              .replace(/\/\//g, `/`)
+          }
+          if (node.children) {
+            node.children = node.children
+              .map(node => addSlugToUrl(node))
+              .filter(Boolean)
           }
-          tocAst.map = addSlugToUrl(tocAst.map)
 
-          toc = hastToHTML(toHAST(tocAst.map))
-        } else {
-          toc = ``
+          return node
         }
-        cache.set(tableOfContentsCacheKey(markdownNode, appliedTocOptions), toc)
-        return toc
+        tocAst.map = addSlugToUrl(tocAst.map)
+
+        toc = hastToHTML(toHAST(tocAst.map, { allowDangerousHTML: true }), {
```

Comment:
My two cents on this PR...

These two lines are the actual change of this commit. It makes reviewing much easier if either the refactorings (which are good, don't get me wrong) were part of a separate commit, or even PR, and this PR would only contain the actual fix (and relevant test changes). When reviewing this PR, the other changes are literally noise...

No need to change it now. More like a suggestion for a future PR :)

Reply:
I went ahead and got rid of the refactoring changes since I was rebasing anyway.